asset-transfer-basic/chaincode-javascript/test/assetTransfer.test.js

```diff
@@ -0,0 +1,251 @@
+'use strict';
+const sinon = require('sinon');
+const chai = require('chai');
+const sinonChai = require('sinon-chai');
+const expect = chai.expect;
+
+const { Context } = require('fabric-contract-api');
+const { ChaincodeStub } = require('fabric-shim');
+
+const AssetTransfer = require('../lib/assetTransfer.js');
+
+let assert = sinon.assert;
+chai.use(sinonChai);
+
+describe('Asset Transfer Basic Tests', () => {
+    let transactionContext, chaincodeStub;
+    beforeEach(() => {
+        transactionContext = new Context();
+
+        chaincodeStub = sinon.createStubInstance(ChaincodeStub);
+        transactionContext.setChaincodeStub(chaincodeStub);
+
+        chaincodeStub.putState.callsFake((key, value) => {
+            if (!chaincodeStub.states) {
+                chaincodeStub.states = {};
+            }
+            chaincodeStub.states[key] = value;
+        });
+
+        chaincodeStub.getState.callsFake(async (key) => {
+            let ret;
+            if (chaincodeStub.states) {
+                ret = chaincodeStub.states[key];
+            }
+            return Promise.resolve(ret);
+        });
+
+        chaincodeStub.deleteState.callsFake(async (key) => {
+            if (chaincodeStub.states) {
+                delete chaincodeStub.states[key];
+            }
+            return Promise.resolve(key);
+        });
+
+        chaincodeStub.getStateByRange.callsFake(async (start, end) => {
+            function* internalGetStateByRange() {
+                if (chaincodeStub.states) {
+                    // Shallow copy
+                    const copied = Object.assign({}, chaincodeStub.states);
+
+                    for (let key in copied) {
+                        yield {value: copied[key]};
+                    }
+                }
+            }
+
+            return Promise.resolve(internalGetStateByRange());
+        });
+    });
+
+    afterEach(() => {
+    });
+
+    describe('Test InitLedger', () => {
+        it('should return error on InitLedger', async () => {
+            chaincodeStub.putState.rejects('failed inserting key');
+            let assetTransfer = new AssetTransfer();
+            try {
+                await assetTransfer.InitLedger(transactionContext);
+                assert.fail('InitLedger should have failed');
+            } catch (err) {
+                expect(err.name).to.equal('failed inserting key');
+            }
+        });
+
+        it('should return success on InitLedger', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.InitLedger(transactionContext);
+            let ret = JSON.parse((await chaincodeStub.getState('asset6')).toString());
+            expect(ret.ID).to.equal('asset6');
```

Comment:
Compare the returned and exepected objects directly using .eql as noted below instead of each key individually

Reply:
Fixed as indicated